acoustic_msgs/msg/SonarImage.msg

```diff
@@ -1,29 +1,56 @@
 #
-# The X-axis is centered in the plane of the fan ("forward"), with Z down,
-# in order to be consistent with the NED convention.
+# Imaging sonars produce either a 1- or 2-D array of sonar beams
+# radiating from a common center.  Data in this message must be in 
+# a North-East-Down coordinate frame.  For the most common case of a 1-D
+# imaging sonar, the array of beams lay on the X-Y plane, and each beam
+# then has an associated azimuth (rotation around the Z-axis).  The
+# definition of the zero azimuth is dependent on the sonar's data format,
+# but for most common sensors, zero azimuth is the middle of the sensor's
+# azimuth range, typically "straight ahead" from the physical sensor.
+# Regardless of the zero azimuth convention, the data much conform with
+# the NED format, with azimuth increasing to starboard.
 #
-# For sonars with a 1D array, the beams lie on the X-Y plane, and
-# and each beam has an azimuth (rotation about Z) value.  Zero azimuth
-# is in the direction of the X-axis, typically "straight out" from the sonar.
-# Elevation is the orthogonal direction (rotation about Y),
-# often termed the "vertical aperture"
+# The azimuth of each beam is given in the vector "azimuth_angles",
+# which allows for unevenly distributed sets of beams.
+#
+# Elevation is defined as rotation around the +Y axis, with negative
+# elevations "pitched down" and positive elevations "pitched up".  1-D
+# sonars have a single elevation value, typically assumed to be zero.
+#
+# 2-D sonars *can* have multiple elevation values.   In the case of 2-D
+# sonar, the "elevation_angles" and "azimuth_angles" are assumed to be
+# fully unzipped s.t. 
+#     len(azimuth_angles)    == number of elevations * number of azimuths
+#     len(elevations_angles) == number of elevations * number of azimuths
+#
+# For each beam, the sonar collects a time sequence of acoustic returns.
+# We assume an estimate of speed of sound, even if only a constant,
+# is available to convert those round trip travel times to ranges in meters,
+# which are quantized into a fixed number of range bins.   Distance from the
+# sensor to the center of each range bin is reported in the "ranges" vector.
+#
+# The measured acoustic return is stored for each range bin, for each 
+# azimuthal beam, in the "intensities[]" array, as an azimuth-major array.
+# That is, the array contains returns for every range in sequence for the
+# first azimuth value, then the returns for every range for the second azimuth
+# value, and so on.   The precise interpretation or units of the intensities 
+# will vary by sensor manufacturer.
 
-# The meaning of the header timestamp relative to the sonar's data
-# acquisition cycle is defined by the sonar driver. It is taken to be
-# a best estimate of the time when the sonar data is valid, which will
-# the timing information available from the sonar hardware.
+# The relationship between the header timestamp and the sonar data
```

Comment:
I think this makes is less clear. 

There is no other place to "specify otherwise" -- this timestamp is what all downstream processing should use, and it will all assume that it is the time that the data is valid.

Reply:
OK, I'll take another crack it...